packages/client/src/client.d.ts

```diff
@@ -11,12 +11,12 @@ declare class Client {
   /**
    * Set default header
    */
-  setDefaultHeader(key: string, value: string): this;
+  setDefaultHeader(key: string | { [s: string]: string }, value ?: string): this;
```

Comment:
I do believe that there is a more advanced way of typing this using custom generics, but I don't have the expertise to do that so I'll likely make another ticket to accomplish that.

The only reason I bring that up is because technically this typing is weaker than it was before, because we now have the ability to supply a `key: string` without a `value`.

I'm not dinging your work, I'm more just articulating a not-so-easy typescript problem for reference elsewhere :)